src/rustdoc_gen.rs

```diff
@@ -123,11 +182,48 @@ pub(crate) enum CrateType<'a> {
     },
 }
 
+#[derive(Debug, Clone, Hash)]
+pub(crate) struct FeatureConfig {
+    pub(crate) base_features: BaseFeatures,
+    pub(crate) extra_features: Vec<String>,
+    pub(crate) ignore_non_existing: bool,
+}
+
+#[derive(Debug, Clone, Hash)]
+pub(crate) enum BaseFeatures {
+    All,
+    Default,
+    Heuristic,
+    None,
+}
+
+impl FeatureConfig {
+    pub(crate) fn default(is_baseline: bool) -> Self {
```

Comment:
This is a very unfortunate name for a function. Users seeing `Foo::default()` will assume it's `Default::default()` which is definitely not the case here. Also consider how `FeatureConfig::default(true)` looks — it very much isn't obvious what that `true` means, and also doesn't seem very default if it's already being configured.

Consider splitting this into separate methods for the `true/false` cases of this input argument.

Reply:
Two separate methods seems like a cool idea.

The function name is a victim of a refactor, when I named it, it didn't have arguments.